spring:
  datasource:
    url: ${DB_URL}
    username: ${DB_USERNAME}
    password: ${DB_PASSWORD}
  jpa:
    hibernate:
      ddl-auto: none
  liquibase:
    change-log: classpath:db/changelog/changelog.xml
  jackson:
    deserialization:
      fail-on-unknown-properties: true
  data:
    web:
      pageable:
        max-page-size: 50
  
  cloud:
    openfeign:
      client:
        config:
          default:
            connect-timeout: 1000
            read-timeout: 5000
    function:
      definition: createPayment
    stream:
      bindings:
        createPayment-in-0:
          destination: passengerPayCardTopic
          content-type: application/json
          group: passengerPayCardCreating
        outCardPaymentCreated:
          destination: cardPaymentCreatedTopic
          content-type: application/json
      kafka:
        binder:
          brokers: "${KAFKA_HOST}:${KAFKA_PORT}"
          enable-observation: true

management:
  endpoints:
    web:
      exposure:
        include: health, info, prometheus
  metrics:
    export:
      prometheus:
        enabled: true
  distribution:
    percentiles-histogram:
      "[http.server.requests]": true
  tracing:
    sampling:
      probability: 1.0
  zipkin:
    tracing:
      endpoint: http://${ZIPKIN_HOST}:${ZIPKIN_PORT}/api/v2/spans

logging:
  level:
    root: info
    liquibase: error

exception:
  handling:
    include: true

resilience4j:
  circuitbreaker:
    configs:
      default:
        failure-rate-threshold: 50
        sliding-window-type: time-based
        sliding-window-size: 100
        minimum-number-of-calls: 10
        permitted-number-of-calls-in-half-open-state: 10
        wait-duration-in-open-state: 20s
        slow-call-duration-threshold: 4s
        slow-call-rate-threshold: 90
        record-exceptions:
          - org.springframework.web.client.HttpServerErrorException
          - java.io.IOException
          - java.util.concurrent.TimeoutException
          - org.springframework.web.client.ResourceAccessException
          - feign.RetryableException
